In [1]:
import pandas as pd
import matplotlib as plt
import numpy as np

In [32]:
file = '../../results/karateclub/pretrained_gnn_reddit_incomplete.csv'
name = file.split('/')[-1].split('.')[0]
df = pd.read_csv(file)

In [33]:
keys = list(df.keys())

In [34]:
relevant_keys = keys[1::3]

In [35]:
print(len(relevant_keys), relevant_keys)

21 ['true-jazz-170 - test_roc_auc', 'laced-monkey-110 - test_roc_auc', 'daily-firefly-121 - test_roc_auc', 'usual-capybara-118 - test_roc_auc', 'tough-shape-164 - test_roc_auc', 'logical-moon-162 - test_roc_auc', 'honest-water-107 - test_roc_auc', 'playful-moon-104 - test_roc_auc', 'lucky-field-105 - test_roc_auc', 'proud-bee-114 - test_roc_auc', 'sage-armadillo-109 - test_roc_auc', 'misty-field-113 - test_roc_auc', 'rosy-eon-119 - test_roc_auc', 'glad-brook-115 - test_roc_auc', 'curious-dream-116 - test_roc_auc', 'dashing-violet-161 - test_roc_auc', 'misty-sound-163 - test_roc_auc', 'super-sound-106 - test_roc_auc', 'eager-mountain-120 - test_roc_auc', 'breezy-bird-171 - test_roc_auc', 'crimson-breeze-173 - test_roc_auc']


In [36]:
rocs = []
for key in relevant_keys:
    rocs += df[key].to_list()
print('%s & $%2.2f \\pm %2.2f$ & $%2.2f \\pm %2.2f$\\\\'%(name, np.mean(rocs)*100, np.std(rocs)*100, np.mean(rocs)*100, (1.96*np.std(rocs)*100)/10))

pretrained_gnn_reddit_incomplete & $83.45 \pm 0.22$ & $83.45 \pm 0.04$\\


transformer_twitch & $72.44 \pm 0.30$ & $72.44 \pm 0.06$ \\
pretrained_transformer_twitch & $71.82 \pm 4.43$ & $71.82 \pm 0.87$\\
gcn_twitch & $72.17 \pm 0.33$ & $72.17 \pm 0.06$ \\
pretrained_gcn_twitch & $72.01 \pm 0.36$ & $72.01 \pm 0.07$\\
wl_twitch & $71.89 \pm 0.31$ & $71.89 \pm 0.06$ \\

transformer_reddit & $84.01 \pm 0.17$ & $84.01 \pm 0.03$\\
pretrained_transformer_reddit & $84.12 \pm 0.16$ & $84.12 \pm 0.03$\\
gcn_reddit & $83.83 \pm 0.17$ & $83.83 \pm 0.03$ \\
pretrained_gcn_reddit_incomplete & $83.45 \pm 0.22$ & $83.45 \pm 0.04$\\
wl_reddit & $83.03 \pm 0.15$ & $83.03 \pm 0.03$ \\

In [37]:
(1.96*0.17)/10

0.03332

In [38]:
len(rocs)

84